packages/expo-file-system/ios/EXFileSystem/EXFileSystem.m

```diff
@@ -675,99 +712,118 @@ - (NSDictionary *)encodingMap
 
 #pragma mark - Internal methods
 
-- (void)_downloadResumableCreateSessionWithUrl:(NSURL *)url withScopedPath:(NSString *)scopedPath withUUID:(NSString *)uuid withOptions:(NSDictionary *)options withResumeData:(NSData * _Nullable)resumeData withResolver:(UMPromiseResolveBlock)resolve withRejecter:(UMPromiseRejectBlock)reject
+- (EXFileSystemSessionType)_importSessionType:(NSNumber * _Nullable)type
 {
-  NSURLSessionConfiguration *sessionConfiguration = [NSURLSessionConfiguration defaultSessionConfiguration];
+  return [type intValue] ?: EXFileSystemBackgroundSession;
+}
+
+- (NSURLSession *)_sessionForType:(EXFileSystemSessionType)type
+{
+  switch (type) {
+    case EXFileSystemBackgroundSession:
+      return _backgroundSession;
+    case EXFileSystemForegroundSession:
+      return _foregroundSession;
+  }
+  return nil;
+}
+
+- (void)_unregisterIfResumableTaskDelegate:(EXSessionTaskDelegate *)exTask
+{
+  if ([exTask isKindOfClass:[EXSessionResumableDownloadTaskDelegate class]]) {
+    EXSessionResumableDownloadTaskDelegate *exResumableTask = (EXSessionResumableDownloadTaskDelegate *)exTask;
+    [_resumableDownloads removeObjectForKey:exResumableTask.uuid];
+  }
+}
+
+- (BOOL)_checkHeadersDictionary:(NSDictionary * _Nullable)headers
+{
+  for (id key in [headers allKeys]) {
+    if (![key isKindOfClass:[NSString class]] || ![headers[key] isKindOfClass:[NSString class]]) {
+      return false;
+    }
+  }
+  
+  return true;
+}
+
+- (NSURLSession *)_createSession:(EXFileSystemSessionType)type
+{
+  NSURLSessionConfiguration *sessionConfiguration;
+  if (type == EXFileSystemForegroundSession) {
+    sessionConfiguration = [NSURLSessionConfiguration defaultSessionConfiguration];
+  } else {
+    sessionConfiguration = [NSURLSessionConfiguration backgroundSessionConfigurationWithIdentifier:[[NSUUID UUID] UUIDString]];
+  }
   sessionConfiguration.requestCachePolicy = NSURLRequestReloadIgnoringLocalCacheData;
   sessionConfiguration.URLCache = nil;
-  
-  __weak typeof(self) weakSelf = self;
+  return [NSURLSession sessionWithConfiguration:sessionConfiguration
+                                       delegate:self
+                                  delegateQueue:[NSOperationQueue mainQueue]];
```

Comment:
Should we use `mainQueue` for this? Does it have performance impact? Shouldn't we use a separate `NSOperationQueue`?

https://github.com/expo/react-native/blob/8131b7bb7b4794e0e7003a6e3d34e1ebe4b8b9bc/Libraries/Network/RCTDataRequestHandler.m#L33

Reply:
I've changed it to `nil` - the system should create a queue for us.